### ISSUE TITLE
Xcode 8 and Swift 2.3?

### ISSUE THREAD

**Issue by dgilperez** [bizz84/SwiftyStoreKit#86]
Hi!

I'm trying to use this framework along with Turbolinks 5, which requires Swift 2.3. I'm using Carthage to include the library, and I can't get it to build. 

`master` branch targets Swift 3, so it won't work. `swift-2-2` branch also fails to build with the following stacktrace: 

```
*** Building scheme "SwiftyStoreKitOSX" in SwiftyStoreKit.xcodeproj
** BUILD FAILED **


The following build commands failed:
	CompileSwift normal x86_64 /path/Carthage/Checkouts/SwiftyStoreKit/SwiftyStoreKit/InAppCompleteTransactionsObserver.swift
	CompileSwift normal x86_64 /path/Carthage/Checkouts/SwiftyStoreKit/SwiftyStoreKit/InAppProductPurchaseRequest.swift
	CompileSwiftSources normal x86_64 com.apple.xcode.tools.swift.compiler
(3 failures)
/path/Carthage/Checkouts/SwiftyStoreKit/SwiftyStoreKit/InAppCompleteTransactionsObserver.swift:80:86: error: cannot convert value of type 'SKPaymentTransactionState' to expected argument type 'Int'
/path/Carthage/Checkouts/SwiftyStoreKit/SwiftyStoreKit/InAppProductPurchaseRequest.swift:109:86: error: cannot convert value of type 'SKPaymentTransactionState' to expected argument type 'Int'
/path/Carthage/Checkouts/SwiftyStoreKit/SwiftyStoreKit/InAppProductQueryRequest.swift:82:10: warning: parameter of 'request(_:didFailWithError:)' has different optionality than expected by protocol 'SKRequestDelegate'
    func request(request: SKRequest, didFailWithError error: NSError?) {
    optional public func request(request: SKRequest, didFailWithError error: NSError)
A shell task (/usr/bin/xcrun xcodebuild -project /path/Carthage/Checkouts/SwiftyStoreKit/SwiftyStoreKit.xcodeproj -scheme SwiftyStoreKitOSX -configuration Release ONLY_ACTIVE_ARCH=NO CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES clean build) failed with exit code 65:
** BUILD FAILED **


The following build commands failed:
	CompileSwift normal x86_64 /path/Carthage/Checkouts/SwiftyStoreKit/SwiftyStoreKit/InAppCompleteTransactionsObserver.swift
	CompileSwift normal x86_64 /path/Carthage/Checkouts/SwiftyStoreKit/SwiftyStoreKit/InAppProductPurchaseRequest.swift
	CompileSwiftSources normal x86_64 com.apple.xcode.tools.swift.compiler
```

So the question is: is SwiftyStoreKit compatible with Swift 2.3, and how to set it up?

Thanks a lot!

**Comment by bizz84** [comment]
Hi there,

I have not added support for Swift 2.3 yet (I should have!), however this should not be too difficult if you need it:

* Fork and clone the [Swift 2.2](https://github.com/bizz84/SwiftyStoreKit/tree/swift-2.2) branch
* Open the project with Xcode 8
* Xcode will ask if you want to convert the project to the latest syntax. Choose "Convert" and when asked select Swift 2.3 as a language
* There should be a screen asking you which targets you want to update. Select them all.
* Xcode should do some chruching, then present a preview window with all the changes. Check that everything looks ok and confirm. Your project should now be ported to Swift 2.3 
* Try to build and run the demo app. There may still be some errors as Xcode conversion sometimes doesn't work 100% but you should be able to resolve these individually.
* Commit your changes.

Once finished, you can send me a pull request. I'll try to review and merge it in a timely manner.

Good luck!

**Comment by bizz84** [comment]
Swift 2.3 support has been added here:
https://github.com/bizz84/SwiftyStoreKit/tree/swift-2.3
Available on pod version 0.4.0

**Comment by dgilperez** [comment]
Ohh nice! 🍺
